### ISSUE TITLE
Some packages attempt to run `make check` while cross compiling

### ISSUE THREAD

**Issue by bgamari** [NixOS/nixpkgs#30437]
## Issue description

Some derivations (e.g. `xz`) force `doCheck = true`, which naturally fails while cross-compiling. Presumably these should instead force `doCheck = stdenv.hostPlatform == stdenv.buildPlatform`? @Ericson2314, thoughts?

### Steps to reproduce

```
$ cat test.nix
let
  crossSystem = {
    config = "arm-unknown-linux-gnueabihf";
    arch = "armv7";
    libc = "glibc";
    withTLS = true;
    openssl.system = "linux-generic32";
    platform = nixpkgsCross.pkgs.platforms.armv7l-hf-multiplatform // {
      baseKernelConfig = "xilinx_zynq_defconfig";
    };
  };

  nixpkgsCross = (import ./nixpkgs) { system = builtins.currentSystem; crossSystem = crossSystem; };
in nixpkgsCross.xz
$ nix-build test.nix
```

## Technical details

* System: Debian 9
* Nix version: `nix-env (Nix) 1.11.8`
* Nixpkgs version: 4d20f425fdd4ab19e720795e6b9c81d02eb3418f
* Sandboxing enabled: No

**Comment by bgamari** [comment]
@Ericson2314 and I discussed this. His plan was to just be explicit, setting `doCheck = stdenv.hostPlatform == stdenv.buildPlatform` where appropriate.

**Comment by bgamari** [comment]
Many of these have been fixed in https://github.com/NixOS/nixpkgs/pull/30883.

**Comment by vcunat** [comment]
Maybe we should make stdenv ignore `doCheck` if cross-compiling (or only in those cases where a different HW is needed).  And probably use `crossAttrs` or something if you _really_ want to run checks even in that case.

**Comment by Ericson2314** [comment]
Hmm that's clever but I was hoping to get rid of `crossAttrs`.

**Comment by vcunat** [comment]
:-) then `doCheckCross` or whatever, but probably not before some package actually manages to pass tests when cross-compiling...

**Comment by Ericson2314** [comment]
Agreed.

**Comment by Ericson2314** [comment]
I still hope #33599 can be done in some fashion, but as far as brevity is concerned, this is fixed by #33603
